invenio_banners/services/service.py

```diff
@@ -101,3 +144,21 @@ def disable_expired(self, identity):
         """Disable expired banners."""
         self.require_permission(identity, "disable")
         self.record_cls.disable_expired()
+
+    def _validate_bool(self, value):
+        import distutils.util
+
+        try:
+            bool_value = distutils.util.strtobool(value)
+        except ValueError:
+            return None
+        return bool(bool_value)
+
+    def _validate_datetime(self, value):
+        from datetime import datetime
+
+        try:
+            datetime_value = datetime.strptime(value, "%Y-%m-%d")
```

Comment:
minor: we can also use the `arrow` library, which is commonly used throughout Invenio (not sure where the dependency is introduced though).